### PR TITLE
[interactive] Fix #17740: `az interactive` fails with `progress_patch() got an unexpected keyword argument 'det'`

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.
Fix https://github.com/Azure/azure-cli/issues/17740
As progress_patch() will raise exception when specifying det=xx, we change progress_patch() to accept key-value pairs.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks

File: src/interactive/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.4.5
++++++
+* Fix issue #17740: "az aks create" fails with "progress_patch() got an unexpected keyword argument 'det'".
```
Comment (jiasli) [inline src/interactive/HISTORY.rst:8]: This issue doesn't only affect `az aks create`:

```suggestion
* Fix #17740: `az interactive` with `progress_patch() got an unexpected keyword argument 'det'`
```

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
Good work. 🚢

**Comment by jiasli** [comment]
Why are chances reverted?

**Comment by Juliehzl** [comment]
> Why are chances reverted?

rebase main branch to make sure correct incremental test detection. Otherwise, there will be some test failures.

**Comment by jiasli** [comment]
You didn't pull before rebasing, causing changes from code review to be lost.

**Comment by Juliehzl** [comment]
> code review ([d67a86d](https://github.com/Azure/azure-cli-extensions/commit/d67a86db3fbdd37126d1e23b83b84df07879a1f3))

Thanks for the check. Add back now.

**Review by jiasli — APPROVED** [review]

**Pull request merged**